exercises/RobertoAssandri/LinkedList/LinkedListProgram.go

```diff
@@ -0,0 +1,266 @@
+package main
+
+import "fmt"
+
+const emptyListMsg = "list has no elements\n"
+
+type Node struct {
+	next *Node
+	value int
+}
+
+type List struct {
+	first *Node
+	last *Node
+}
+
+func InsertFirst(list *List, number int){
+	node := Node{
+		value: number,
+	}
+
+	if list.first == nil{
+		list.last = &node
+	} else {
+		node.next = list.first
+	}
+
+	list.first = &node
+}
+
+func PrintList(list List){
+	if list.first != nil{
+		PrintListRec(*list.first)
+	} else {
+		fmt.Print(emptyListMsg)
+	}
+}
+
+func PrintListRec(node Node){
+	if node.next == nil{
+		fmt.Printf("%d\n", node.value)
+	}else if node.next != nil{
+		fmt.Printf("%d - ", node.value)
+		PrintListRec(*node.next)
+	}
+}
+
+func ReturnFirst(list List) (int, error){
+	if list.first != nil{
+		return list.first.value, nil
+	}
+
+	return 0, fmt.Errorf(emptyListMsg)
+}
+
+func PrintElement(value int, err error){
+	if err != nil{
+		fmt.Printf("Error %s\n", err)
+	} else {
+		fmt.Printf("The element is %d\n", value)
+	}
+}
+
+func DeleteFirst(list *List){
+	if list.first != nil{
+		list.first = list.first.next
+	} else{
+		fmt.Print(emptyListMsg)
+	}
+}
+
+func Contains(list List, value int) (bool, error){
+	if list.first == nil{
+		return false, fmt.Errorf(emptyListMsg)
+	} else {
+		return ContainsRec(*list.first, value), nil
+	}
+}
+
+func ContainsRec(node Node, value int) bool{
+	if node.value == value{
+		return true
+	} else if node.next == nil{
+		return false
+	} else {
+		return ContainsRec(*node.next, value)
+	}
+}
+
+func CopyList(list List) List{
+	if list.first == nil{
+		return List{}
+	} else {
+		copyList := List{}
+		copyList.first = CopyListRec(list.first, &copyList)
+		return copyList
+	}
+}
+
+func CopyListRec(original *Node, copyList *List) *Node {
+	copyNode := Node {}
+	copyNode.value = original.value
+	if original.next != nil{
+		copyNode.next = CopyListRec(original.next, copyList)
+	} else {
+		copyList.last = &copyNode
+	}
+	return &copyNode
+}
+
+func FreeList(list *List){
```

Comment:
Could be resolved as?
func FreeList(list *List){
	list.first = nil
	list.last = nil
}